source/vistas/ui/windows/linreg_dialog.py

```diff
@@ -91,14 +405,41 @@ def plotLinReg(self, iv=None, dv=None):
         ols = sm.OLS(dv_data, sm.add_constant(iv_data))
         result = ols.fit()
 
+        self.createGraph(iv, dv, my_data, result)
+
+        #Only draw tables if variables have been changed
+        if self.reset:
+            self.createTable(iv, result)
+
+    def createGraph(self, iv=None, dv=None, my_data=None, result=None):
+        try:
+          self.fig.delaxes(self.ax)
+        except:
```

Comment:
Catching and passing _all_ exceptions is almost always undesirable. The reason is that any number of issues could occur that cause a silent failure (no message), that is hard to track down. Even programming errors (such as a typo in the variable name, resulting in a variable that doesn't exist) can sneak by.

Instead, if you know there's a specific error that is okay to ignore, catch that error specifically. For example, if you know that `delaxes` raises `KeyError` when the axis doesn't exist, and you want to simply move on without error when that happens, then:

```python
try:
    self.fig.delaxes(self.ax)
except KeyError:
    pass
```

Now the key errors that you expect will be ignored, but another, unexpected error will still cause an exception.